test/gpu/octree/test_knn_search.cpp

```diff
@@ -127,12 +129,15 @@ TEST(PCL_OctreeGPU, exactNeighbourSearch)
     //search GPU shared
     {
         pcl::ScopeTime time("1nn-gpu");
-        octree_device.nearestKSearchBatch(queries_device, k, result_device);
+        octree_device.nearestKSearchBatch(queries_device, k, result_device, result_sqr_distances);
     }
 
     std::vector<int> downloaded, downloaded_cur;
     result_device.data.download(downloaded);
-                 
+
+    std::vector<float> downloaded_sqr_dists, downloaded_sqr_dists_cur;
```

Comment:
downloaded_sqr_dists_cur is not being used here. declare it only when you use it.